site/cypress/e2e/Cypress-Feature/live_chat.spec.js

```
@@ -0,0 +1,340 @@
+const name1 = 'Quinn Instructor';
+const name2 = 'Joe S.';
+
+const getChatroom = (title) => {
+    return cy.get('[data-testid="chatroom-item"]').filter((index, el) => {
+        return Cypress.$(el).find('[data-testid="chatroom-title"]').text().trim() === title;
+    }).first();
+};
+
+const startChatSession = (title) => {
+    chatroomExists(title).then((exists) => {
+        if (exists) {
+            getChatroom(title).find('[data-testid="enable-chatroom"]').first().click();
+        }
+    });
+};
+
+const endChatSession = (title) => {
+    chatroomExists(title).then((exists) => {
+        if (exists) {
+            cy.get('body').then(($body) => {
+                const chatroom = $body.find(`[data-testid="chatroom-item"]:has([data-testid="chatroom-title"]:contains("${title}"))`);
+                const disableBtn = chatroom.find('[data-testid="disable-chatroom"]');
+
+                if (disableBtn.length > 0) {
+                    cy.window().then((win) => {
+                        cy.stub(win, 'confirm').callsFake((confirmText) => {
+                            expect(confirmText).to.contain('close the chatroom');
+                            return true;
+                        });
+                    });
+
+                    getChatroom(title).find('[data-testid="disable-chatroom"]').first().click();
+                }
+            });
+        }
+    });
+};
+
+const toggleLiveChat = (enableChat) => {
+    cy.visit(['sample', 'config']);
+    return cy.get('input#chat-enabled').first().then(($checkbox) => {
+        const isCurrentlyEnabled = $checkbox.is(':checked');
+        if (isCurrentlyEnabled !== enableChat) {
+            cy.wrap($checkbox).click();
+        }
+    });
+};
+
+const chatroomExists = (title) => {
+    return cy.get('body').then(($body) => {
+        if ($body.find('[data-testid="chatroom-item"]').length === 0) {
+            return false;
+        }
+
+        return cy.get('[data-testid="chatroom-item"]').then(($rows) => {
```

These helper methods appear to introduce Cypress anti-patterns by breaking command chains and bypassing built-in retry logic. If the chatroom is supposed to exist, we should fail early.

```js
const assertChatroomExists = (title) => {
    cy.get('[data-testid="chatroom-title"]')
        .contains(title)
        .should('exist'); // Auto-retries and fails test if not found
};
```

---

Failing on non-existence for non-cleanup methods and ignoring otherwise sounds fine.